### PR TITLE
azure-json, fix package-private setter type

### PR DESCRIPTION
**Pull request by XiaofeiCao**
- fix https://github.com/Azure/autorest.java/issues/2805, change package private setter parameter type to `clientType`.
- also make immutable model consistent

### CODE REVIEW

File: javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java [hunk 1]
```diff
@@ -202,11 +201,7 @@ public final void write(ClientModel model, JavaFile javaFile) {
                     methodBlock -> addGetterMethod(propertyWireType, propertyClientType, property, treatAsXml,
                             methodBlock, settings));
 
-                // The model is immutable output only if and only if the immutable output model setting is enabled and
-                // the usage of the model include output and does not include input.
-                boolean immutableOutputOnlyModel = immutableOutputModel && ClientModelUtil.isOutputOnly(model);
-
-                if (ClientModelUtil.hasSetter(property, settings) && !immutableOutputOnlyModel) {
+                if (ClientModelUtil.hasSetter(property, settings) && !immutableModel) {
```
Comment (XiaofeiCao) [inline javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java:204]: Change here to make immutable logic consistent between class and properties.

**Review by weidongxu-microsoft — APPROVED** [review]

File: javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java (hunk 1 above)
Comment (XiaofeiCao) [reply javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java:204]: Yes, this happens in vanilla.

File: javagen/src/main/java/com/azure/autorest/util/ClientModelUtil.java [hunk 2]
```diff
@@ -755,4 +755,9 @@ public static boolean requireOperationLocationPollingStrategy(CodeModel codeMode
     public static boolean isMultipartModel(ClientModel model) {
         return model.getSerializationFormats().contains(KnownMediaType.MULTIPART.value());
     }
+
+    public static boolean noUsage(ClientModel model) {
```
[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply javagen/src/main/java/com/azure/autorest/util/ClientModelUtil.java:759]: `hasNoUsage` seems better.

[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply javagen/src/main/java/com/azure/autorest/util/ClientModelUtil.java]: changed

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**